src/lib.rs

```diff
@@ -182,8 +218,10 @@ impl Datadog {
 
     pub(crate) fn do_incr(&self, metric: impl AsRef<str>, tags: impl IntoIterator<Item = String>) {
```

Comment:
I think it's overkill and we should keep this simple but we could replace/generate all of the do_x(do_incr, do_decr, do_count, etc.) functions with a proc macro

Reply:
hmm perhaps but I wouldn't in this pr

I'm not sure that would be totally worthwhile, unless you could generalize it... the pattern is "if let Some(x) and x.y then {block involving x}" which I suppose could be useful elsewhere, but I am not sure.